cmake/compiler/compiler_flags_template.cmake

```diff
@@ -68,6 +68,10 @@ set_property(TARGET compiler-cpp PROPERTY dialect_cpp2b)
 # Flag for disabling strict aliasing rule in C and C++
 set_compiler_property(PROPERTY no_strict_aliasing)
 
+# Extra warnings options for twister run
+set_property(TARGET compiler PROPERTY cc_extra_warning_options)
```

Comment:
the intention with the properties is to have a common property name which then can be mapped to whatever flag a given toolchain is using to achieve such behavior.

In this case an appropriate name for the property would be probably be `warning_as_errors`, for both C, asm, and linker.

Reply:
Done